### ISSUE TITLE
Mylyn Task Editor Integration

### ISSUE THREAD

**Issue by BeckerFrank** [eclipse-mylyn/org.eclipse.mylyn.docs#131]
In https://help.eclipse.org/latest/topic/org.eclipse.mylyn.wikitext.help.ui/help/Task-Editor-Integration.html?cp=59_1_2 I found the Task Editor Integration, but it looks like this feature is disabled / removed. Any plans to bring this back?

**Comment by ruspl-afed** [comment]
Why not?

**Comment by akurtakov** [comment]
I have removed it when Mylyn task editor dropped from simrel. I don't mind if anyone brings it back but I would not have the time for that.

**Comment by akurtakov** [comment]
FWIW, code is still there https://github.com/eclipse-mylyn/org.eclipse.mylyn.docs/tree/master/wikitext/ui/org.eclipse.mylyn.wikitext.tasks.ui but not added in the pom.xml to try compiling it.

**Comment by BeckerFrank** [comment]
@akurtakov: Where can I found instructions to setup a Mylyn Docs Eclipse instance?
I can try to bring org.eclipse.mylyn.wikitext.tasks.ui back.

**Comment by merks** [comment]
I was trying to create an Oomph setup but I got completely confused by the structure. I too was planning to ask how one manually sets up an error free workspace. But then a dozen other things took precedence.

**Comment by akurtakov** [comment]
What I've always been doing is import all projects from git and then close not build ones. Having such a workspace for long time doesn't require much changes from my side. 
Please note that my interest in wikitext is limited to having it working as markdown renderer in lsp4e. This required a lot of effort to keep builds working but other improvements are for someone else to handle.

**Comment by merks** [comment]
So maybe at some point it would make sense to delete the not-built-ones?   Certainly it makes sense to automate whatever the steps are to produce a workable workspace, because the definition of the "not build ones" is obviously far less than clear.

Note that I'm asking you to do anything!

**Comment by akurtakov** [comment]
Not build ones are Mylyn Tasks and context integration AFAIK so maybe it's good that I haven't deleted them back in the days when Mylyn dropped from simrel. 
Overall, I totally agree with you that projects shouldn't carry so much deadweight.

**Comment by merks** [comment]
FYI, this is as far as I got in the automation:

![image](https://github.com/eclipse-mylyn/org.eclipse.mylyn.docs/assets/208716/fdce5aa3-a453-4f4b-b902-f9bfe20fa41f)

This is the result:

![image](https://github.com/eclipse-mylyn/org.eclipse.mylyn.docs/assets/208716/c3325aaf-1b10-4787-ac04-edf1e8c618f1)

And then I got distracted by other activities...

**Comment by BeckerFrank** [comment]
@akurtakov: can you share some information how you setup your workspace. @merks or I can the try to build a MylynDocs.setup to automate the steps.

**Comment by akurtakov** [comment]
It's all done manual:
* Import all projects from the git repo
* Set the target platform to the one shipped in the git repo
* Close plugins that are not built by the maven build

**Comment by BeckerFrank** [comment]
> It's all done manual:
> 
> * Import all projects from the git repo
> * Set the target platform to the one shipped in the git repo
> * Close plugins that are not built by the maven build

I did this but have 904 errors. I think that I missed something because I get errors like.
Bundle 'org.eclipse.mylyn.wikitext' cannot be resolved from other MANIFEST.MF files.

In my Eclipse Instance I use the m2e plugin. Can this cause the errors?

**Comment by akurtakov** [comment]
you have to compile once at the console to get the manifests generated

**Comment by merks** [comment]
Sorry I forgot to share the rough setup.  I’ll try to remember tomorrow.

**Comment by BeckerFrank** [comment]
OK now I have a problem with the mvn compile.

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (default) on project org.eclipse.mylyn.wikitext.help.ui: An Ant BuildException has occured: Problem: failed to create task or type wikitext-to-eclipse-help
[ERROR] Cause: The name is undefined.
[ERROR] Action: Check the spelling.
[ERROR] Action: Check that any custom tasks/types have been declared.
[ERROR] Action: Check that any <presetdef>/<macrodef> declarations have taken place.
[ERROR] 
[ERROR] around Ant part ...<wikitext-to-eclipse-help markupLanguage="org.eclipse.mylyn.wikitext.textile.TextileLanguage" multipleOutputFiles="true" helpPrefix="help" navigationImages="true">... @ 17:168 in /Users/frank/eclipse/mylyn-docs-master/git/org.eclipse.mylyn.docs/wikitext/ui/org.eclipse.mylyn.wikitext.help.ui/target/antrun/build-main.xml

**Comment by BeckerFrank** [comment]
@akurtakov, @merks, @ruspl-afed What do you think of the following option to bring back the task editor integration?

We can do the following.

1. add the Wikitext SDK from the Mylyn Docs update site to the target platform
2. move the org.eclipse.mylyn.wikitext.tasks.ui project to the org.eclipse.mylyn Git repository
3. change the plugin version to 4.1.0
4. Add the Mylyn Docs updatesite as a repository reference

Thougths?

**Comment by akurtakov** [comment]
So you would like to inverse the dependency chain. Wikitext to be required by Mylyn rather than wikitext requiring Mylyn. I don't see problem with that.

**Comment by merks** [comment]
Is merging the repos an option?

**Comment by ruspl-afed** [comment]
I'm not sure that we can just merge these repositories, since Mylyn Docs repository uses very specific way of building bundles.
Moving `org.eclipse.mylyn.wikitext.tasks.ui` from Mylyn Docs to Mylyn looks much more realistic to me.
@BeckerFrank have you tried your plan locally?

**Comment by akurtakov** [comment]
The way Mylyn Docs builds is really weird - when I first looked into it, it was Maven build forking into several extra maven builds each of them using different maven plugin versions and relying on artifacts being built/installed into local p2 repos. 
I managed to get it to the single Maven reactor which we have now but this is still far from optimal and plain ugly and fragile. One more iteration of simplification is highly due if someone can spend the time on it.

**Comment by BeckerFrank** [comment]
> I'm not sure that we can just merge these repositories, since Mylyn Docs repository uses very specific way of building bundles. Moving `org.eclipse.mylyn.wikitext.tasks.ui` from Mylyn Docs to Mylyn looks much more realistic to me. @BeckerFrank have you tried your plan locally?

Yes, I have tried this locally. I noticed that Mylyn Docs does not have a latest P2 update site, so I used version [3.0.48] (https://download.eclipse.org/mylyn/docs/releases/3.0.48).



> The way Mylyn Docs builds is really weird - when I first looked into it, it was Maven build forking into several extra maven builds each of them using different maven plugin versions and relying on artifacts being built/installed into local p2 repos. I managed to get it to the single Maven reactor which we have now but this is still far from optimal and plain ugly and fragile. One more iteration of simplification is highly due if someone can spend the time on it.

This is why I came up with the idea to move the one project from Mylyn Docs to Mylyn.
